src/ShadyCSS.js

```diff
@@ -146,52 +149,74 @@ export let ShadyCSS = {
       }
     }
     let styleInfo = StyleInfo.get(host);
-    let hasApplied = Boolean(styleInfo);
     if (!styleInfo) {
       styleInfo = this._prepareHost(host);
     }
     if (overrideProps) {
       styleInfo.overrideStyleProperties =
         styleInfo.overrideStyleProperties || {};
-      Object.assign(styleInfo.overrideStyleProperties, overrideProps);
+      assign(styleInfo.overrideStyleProperties, overrideProps);
+      this._forceRecalc = true;
     }
-    if (this.nativeCss) {
-      let template = templateMap[is];
-      if (template && template.__applyShimInvalid && template._style) {
-        // update template
-        ApplyShim.transformRules(template._styleAst, is);
-        template._style.textContent = StyleTransformer.elementStyles(host, styleInfo.styleRules);
-        // update instance if native shadowdom
-        if (this.nativeShadow) {
-          let style = host.shadowRoot.querySelector('style');
-          style.textContent = StyleTransformer.elementStyles(host, styleInfo.styleRules);
+    let template = templateMap[is];
+    if (this._forceRecalc || styleInfo.shouldRecalculate || template._applyShimInvalid) {
```

Comment:
let's also bail out of this function early if we detect the element does not use or produce `apply` rules and native properties are in use.